contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java

```diff
@@ -448,7 +448,11 @@ public int next() {
     }
 
     if (nonExistentColumnsProjection && recordCount > 0) {
-      JsonReaderUtils.ensureAtLeastOneField(vectorWriter, getColumns(), schema, allTextMode, Collections.emptyList());
+      if (schema != null && !schema.isEmpty()) {
```

Comment:
You can invert condition to avoid negation:
```
if (schema == null || schema.isEmpty()) {
```

Reply:
Thanks, done.